server/http_integration_test.go

```diff
@@ -458,6 +465,117 @@ func TestHTTPServer_XFHHeader(t *testing.T) {
 	cleanup(shutdown, t)
 }
 
+func TestHTTPServer_Gzip(t *testing.T) {
+	client := newClient()
+
+	confPath := path.Join("testdata/integration", "files/02_couper.hcl")
+	shutdown, _ := newCouper(confPath, test.New(t))
+
+	t.Run("Test", func(subT *testing.T) {
+		helper := test.New(subT)
+
+		req, err := http.NewRequest(http.MethodGet, "http://example.org:9898/c", nil)
+		helper.Must(err)
+
+		req.Header.Set("Accept-Encoding", "br, gzip")
+		req.Host = "example.org"
+		res, err := client.Do(req)
+		helper.Must(err)
+
+		if ce := res.Header.Get("Content-Encoding"); ce != "gzip" {
+			t.Errorf("Expected CE <gzip>, given: %s", ce)
+		}
+		if cl := res.Header.Get("Content-Length"); cl != "70" {
+			t.Errorf("Expected CL <70>, given: %s", cl)
+		}
+		if vr := res.Header.Get("Vary"); vr != "Accept-Encoding" {
+			t.Errorf("Expected Vary <Accept-Encoding>, given: %s", vr)
+		}
+
+		var src io.Reader
+
+		src, err = gzip.NewReader(res.Body)
+		helper.Must(err)
+
+		resBytes, err := ioutil.ReadAll(src)
+		helper.Must(err)
+
+		srcBytes, err := ioutil.ReadFile(filepath.Join(testWorkingDir, "testdata/integration/files/htdocs_c/index.html"))
+		helper.Must(err)
+
+		if !bytes.Equal(resBytes, srcBytes) {
+			t.Errorf("Expected \n%s\ngiven\n%s", srcBytes, resBytes)
+		}
+	})
+
+	cleanup(shutdown, t)
+}
+
+func TestServer_DisableCompression(t *testing.T) {
```

Comment:
Both testcases (`TestServer_DisableCompression`, `TestServer_ModifyAcceptEncoding `) are no integration tests since they are using the proxy (backend) directly. So they should live in a `proxy_test` file or the integration test backend needs to be extended (new route) and the tests requires a change. Since the "origin" test the same thing, maybe a table driven test would be a good fit in general. Reading and extracting an origins gzip response via proxy (backend) is not tested here.